src/shared/extensionUtilities.ts

```diff
@@ -364,3 +364,53 @@ export function getComputeRegion(): string | undefined {
 
     return computeRegion
 }
+
+/**
+ * Provides an {@link vscode.Event} that is triggered when the user interacts with the extension.
+ * This will help to be notified of user activity.
+ */
+export class ExtensionUserActivity {
+    private activityEvent = new vscode.EventEmitter<void>()
+    /** The event that is triggered when the user interacts with the extension */
+    onUserActivity = this.activityEvent.event
+
+    /** To be efficient we re-used instances that already have the same throttle delay */
+    private static instances: { [delay: number]: ExtensionUserActivity } = {}
+
+    /**
+     * @param delay The delay until a throttled user activity event is fired in milliseconds.
+     * @param activityEvents The events that trigger the user activity.
+     * @returns
+     */
+    static instance(delay: number = 500, activityEvents = ExtensionUserActivity.activityEvents): ExtensionUserActivity {
+        if (!ExtensionUserActivity.instances[delay]) {
+            const newInstance = new ExtensionUserActivity(delay, activityEvents)
+            ExtensionUserActivity.instances[delay] = newInstance
+        }
+        return ExtensionUserActivity.instances[delay]
+    }
+
+    private constructor(delay: number, private readonly activityEvents: vscode.Event<any>[]) {
+        const throttledEvent = _.throttle(() => this.activityEvent.fire(), delay)
```

Comment:
Does `_.throttle` "collapse" redundant events? Don't want to merely spread out the triggers, but collapse them.

Reply:
Yup it collapses them in to 1 event firing. If multiple vscode events trigger at a similar time within the `delay`, all successive calls will do nothing. So only 1 event firing per `delay`